### PR TITLE
Add custom user model

### PR DESCRIPTION
**Pull request by ChristianLapinig**
TL;DR - I added abilities to register, login, and retrieve users. Below is what I did
* Added custom User model that saves a users email, password, first & last name, and phone number
* Add ability to register users using the fields in the user model
* Add ability to login users that returns a jwt token
* Add ability to authenticate by checking the 'Authorization' header from a HTTP request
* Add user field to Person model (Person essentially becomes a user profile class)

Here are the tests I ran in Postman:
Screenshot
Screenshot 2
Screenshot 3

Even though this works, this breaks the db_populate.py script. If this breaks other parts of the application let me know. I can work on this and make a new pull request.

### CODE REVIEW

**Comment by XZ7QN0** [comment]
Just out of curious @pv-gitjob, were you able to login at all on the API page? I tried logging in with a newly created account and was not able to login at all.

**Comment by pv-gitjob** [comment]
@XZ7QN0 I was able to use an admin account to see the api

**Comment by pv-gitjob** [comment]
@XZ7QN0 You're right, there is 2 users. I guess I didn't notice because the users I created before this one still works. But the API part does work, but it uses the user that I already had.
Screenshot (13)

**Comment by ChristianLapinig** [comment]
@XZ7QN0 are you getting a 403 error when you're trying to hit the /api/ endpoint? It's most likely because an `Authorization` header needs to be present with `Token <jwt-token-here>` as the value. I noticed the two Users groups under Accounts on the admin page. I can continue to work these out. I think the main thing is linking a `User` to a `Person` once a user is created. From there we can figure out how to get the rest of the models to work with `Person`. If someone wants to help me out that would be great. Thanks for getting eyes on this guys.

**Comment by ChristianLapinig** [comment]
@XZ7QN0 @pv-gitjob I fixed the User and Person model and the admin stuff related to each model. Here's what I did:
- Pass models.Model to Person instead of AbstractUser and create a OneToOne relationship between the User model and the Person model
- Create an update and add user form so users can be created and updated in the admin console
- Add rest_framework.authentication.SessionAuthentication to "bypass" Authorization header. Basically everyone can login with a superuser account now in the API views

How are you guys specifically associating the Person models with the rest of the apps?

**Comment by XZ7QN0** [comment]
The `Person` model is being referenced to `Tutor`, `Student`, and `Jobs` so far as a `model.ForeignKey` field, so the idea is that a `Person` will either have made a job offer, is a tutor, or is a student. The linkage of `Person` to `User` happens in the `Person` model, where the creation of a `Person` should automatically be linked to the `User` invoking that creation.

 i.e. if I'm a user with basic information, changing my settings to add information that's contained in the `Person` model should automatically be linked to me, and I shouldn't be able to create another instance of a `Person` object.

**Comment by ChristianLapinig** [comment]
Sounds good. Thanks @XZ7QN0 and @pv-gitjob!

**Comment by ChristianLapinig** [comment]
Looks good to me. We have to make some changes in the populate script to make it work with the User and Person changes
